pkg/pubsub/adapter/adapter.go

```diff
@@ -18,253 +18,218 @@ package adapter
-				return fmt.Errorf("failed to create transformer cloudevent client: %w", err)
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				a.logger.Warn("Failed to close response body", zap.Error(err))
 			}
-		}
-	}
+		}()
 
-	return a.inbound.StartReceiver(ctx, a.receive)
-}
+		a.reporter.ReportEventCount(args, resp.StatusCode)
 
-func (a *Adapter) receive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
-	logger := logging.FromContext(ctx).With(zap.Any("event.id", event.ID()), zap.Any("sink", a.Sink))
+		if resp.StatusCode/100 != 2 {
+			a.logger.Error("Event delivery failed", zap.Int("StatusCode", resp.StatusCode))
+			msg.Nack()
+			return
+		}
 
-	// TODO Name and ResourceGroup might cause problems in the near future, as we might use a single receive-adapter
-	//  for multiple source objects. Same with Namespace, when doing multi-tenancy.
-	args := &ReportArgs{
-		Name:          a.Name,
-		Namespace:     a.Namespace,
-		EventType:     event.Type(),
-		EventSource:   event.Source(),
-		ResourceGroup: a.ResourceGroup,
-	}
+		respMsg := cehttp.NewMessageFromHttpResponse(resp)
+		if respMsg.ReadEncoding() == binding.EncodingUnknown {
+			// No reply
+			msg.Ack()
+			return
+		}
 
-	var err error
-	// If a transformer has been configured, then transform the message.
-	// Note that this path in the code will be executed when using the receive adapter as part of the underlying Channel
-	// of a Broker. We currently set the TransformerURI to be the address of the Broker filter pod.
-	// TODO consider renaming transformer as it is confusing.
-	if a.transformer != nil {
-		transformedCTX, transformedEvent, err := a.transformer.Send(ctx, event)
-		rtctx := cloudevents.HTTPTransportContextFrom(transformedCTX)
+		// If there was a reply, we need to send it to the sink.
+		// We then overwrite the initial event we sent.
+		event, err = binding.ToEvent(ctx, respMsg)
 		if err != nil {
-			logger.Errorf("error transforming cloud event %q", event.ID())
-			a.reporter.ReportEventCount(args, rtctx.StatusCode)
-			return err
-		}
-		if transformedEvent == nil {
-			// This doesn't mean there was an error. E.g., the Broker filter pod might not return a response.
-			// Report the returned Status Code and return.
-			logger.Debugf("cloud event %q was not transformed", event.ID())
-			a.reporter.ReportEventCount(args, rtctx.StatusCode)
-			return nil
+			a.logger.Error("Failed to convert response message to event",
+				zap.Any("response", respMsg), zap.Error(err))
+			msg.Ack()
+			return
 		}
-		// Update the event with the transformed one.
-		event = *transformedEvent
+
 		// Update the tracing information to use the span returned by the transformer.
-		ctx = trace.NewContext(ctx, trace.FromContext(transformedCTX))
+		// TODO how to get the span returned by the transformer?
+		ctx = trace.NewContext(ctx, trace.FromContext(ctx))
```

Comment:
@Harwayne @ian-mi 
I think this is where I'm messing up... Previously we had a context returned and we created a new context with the returned span... not sure what is the proper way of doing this with the new cev2 sdk

Reply:
ok, thanks for the pointer Ian, added some stuff. Let's see if that works. 
Otherwise I'll take a look next week